oidc-exchange.py

```diff
@@ -162,6 +175,27 @@ def _get(name: str) -> str:  # noqa: WPS430
     )
 
 
+def event_is_third_party_pr() -> bool:
+    # Non-`pull_request` events cannot be from third-party PRs.
+    if os.getenv("GITHUB_EVENT_NAME") != "pull_request":
+        return False
+
+    if event_path := os.getenv("GITHUB_EVENT_PATH"):
+        try:
+            event = json.loads(Path(event_path).read_text())
+            try:
+                return event["pull_request"]["head"]["repo"]["fork"]
+            except KeyError:
+                return False
+        except json.JSONDecodeError:
+            debug("unexpected: GITHUB_EVENT_PATH does not contain valid JSON")
+            return False
+
+    # No GITHUB_EVENT_PATH indicates a weird GitHub or runner bug.
+    debug("unexpected: no GITHUB_EVENT_PATH to check")
+    return False
```

Comment:
NB: I used `debug(...)` rather than `die(...)` here, since this function is only called from an error path -- I think masking the error here would likely cause even more confusion. But maybe there's a better idea/solution :slightly_smiling_face:

Reply:
Thanks for this note! I don't have better a refactoring suggestion for this right now. Maybe sometime...